charts/spire/charts/spiffe-oidc-discovery-provider/templates/tests/test-keys.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "spiffe-oidc-discovery-provider.fullname" . }}-test-keys"
+  namespace: {{ include "spiffe-oidc-discovery-provider.namespace" . }}
+  labels:
+    {{- include "spiffe-oidc-discovery-provider.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  securityContext:
+    {{- toYaml .Values.podSecurityContext | nindent 4 }}
+  serviceAccountName: {{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }}-test
+  initContainers:
+    - name: jwt
+      image: {{ template "spire-lib.image" (dict "image" .Values.tools.kubectl.image "global" .Values.global) }}
+      command:
+      - sh
+      args:
+        - -cx
+        - |
+          statefulsetname=$(kubectl -n {{ include "spiffe-oidc-discovery-provider.namespace" . }} get statefulset -o=jsonpath='{.items[0].metadata.name}')
+          kubectl -n {{ .Release.Namespace }} exec -q statefulset/$statefulsetname -- spire-server jwt mint -spiffeID spiffe://{{ include "spire-lib.trust-domain" . }}/ns/{{ .Release.Namespace }}/sa/{{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }} -audience spire > /data/token.jwt
```

Comment:
Hmm.... rather then going through the backdoor via server, which is a bit of an unusual path, would it work better to use the agent? Three wouldn't need the role/rolebinding/sa/etc then.

Reply:
what image/command would work. The command needs to push a jwt into a file

Reply:
```
k -n spire-system exec -it spire-agent-pgg2q -- /opt/spire/bin/spire-agent api fetch jwt -socketPath /run/spire/agent-sockets/spire-agent.sock -audience boom -spiffeID spiffe://example.org/ns/spire-system/sa/spire-spiffe-oidc-discovery-provider
Defaulted container "spire-agent" out of: spire-agent, init (init), fsgroupfix (init)
rpc error: code = PermissionDenied desc = no identity issued
command terminated with exit code 1
```

Reply:
If the test ran the spire-agent container (so you have a spire client) along with the csi driver volume mount, so it can access the local spire-agent, something like this might work:
```
/opt/spire/bin/spire-agent api fetch jwt
```